src/sentry/api/endpoints/organization_member_team_details.py

```diff
@@ -78,25 +88,19 @@ def _can_delete(self, request: Request, member, organization, team_slug):
         if request.user.id == member.user_id:
             return True
 
-        if self._can_admin_team(request, organization, team_slug):
+        if self._can_admin_team(request, team):
             return True
 
         return False
 
-    def _can_admin_team(self, request: Request, organization, team_slug):
-        global_roles = [r.id for r in roles.with_scope("org:write") if r.is_global]
-        team_roles = [r.id for r in roles.with_scope("team:write")]
-
-        # must be a team admin or have global write access
-        return OrganizationMember.objects.filter(
-            Q(role__in=global_roles)
-            | Q(organizationmemberteam__team__slug=team_slug, role__in=team_roles),
-            organization=organization,
-            user__id=request.user.id,
-            user__is_active=True,
-        ).exists()
+    def _can_admin_team(self, request: Request, team: Team) -> bool:
+        return (request.access.has_global_access and request.access.has_scope("org:write")) or (
+            team in request.access.teams and request.access.has_team_scope(team, "team:write")
+        )
```

Comment:
@maheskett I'd like to be sure that this is equivalent to the intended behavior from https://github.com/getsentry/sentry/pull/16389, to the extent you can remember 😄. In particular, I had to be careful about `test_admin_cannot_remove_member` passing. When running that test, the four checks' values are as follows:
```python
        has_global_access = request.access.has_global_access  # True
        has_org_scope = request.access.has_scope("org:write")  # False
        is_in_team = team in request.access.teams  # True
        has_team_scope = request.access.has_team_scope(team, "team:write")  # False
```
where the return value is equivalent to `(has_global_access and has_org_scope) or (is_in_team and has_team_scope)`, which we expect to be false.

As an implementation note, we seem to need `team in request.access.teams` separately from `has_team_scope`. The `has_team_scope` method requires "team access" to return true, but allows `has_global_access` in place of actual team membership. Requiring actual team membership seems to pretty clearly be the intent of the original code and unit test. There are some `Access` subclasses where `.teams` is always empty (such as `SystemAccess`), but in the absence of an org ID the original check wouldn't have applied anyway.

I'm not 100% clear on the need for `if r.is_global` check in the original (all tests pass without it) but I'm pretty sure that `request.access.has_global_access and` makes it equivalent. `has_global_access` would additionally be true in cases where `organization.flags.allow_joinleave` is set, but I believe that would be correct behavior anyway.

Reply:
@RyanSkonnord It's been a while but this looks correct to me. I'm not sure `request.access.has_global_access and` is needed either.

Reply:
Yeah, the org-level scope check should suffice I think. I'm going to take it out.